src/app/date-picker/date-picker.directive.ts

```diff
@@ -1,28 +1,65 @@
-import {CalendarType} from '../../../bin/common/types/calendar-type';
+import {CalendarType} from './../common/types/calendar-type';
 import {IDatePickerDirectiveConfig} from './date-picker-directive-config.model';
 import {DatePickerDirectiveService} from './date-picker-directive.service';
 import {IDpDayPickerApi} from './date-picker.api';
 import {DatePickerComponent} from './date-picker.component';
-import {ChangeDetectorRef, OnChanges, SimpleChanges} from '@angular/core';
 import {ElementRef} from '@angular/core';
 import {OnInit} from '@angular/core';
 import {ViewContainerRef} from '@angular/core';
-import {ComponentFactoryResolver, ComponentRef, Directive, HostListener, Input} from '@angular/core';
+import {ComponentFactoryResolver, Directive, HostListener, Input} from '@angular/core';
 import {NgModel} from '@angular/forms';
 
 @Directive({
   exportAs: 'dpDayPicker',
   providers: [DatePickerDirectiveService],
   selector: '[dpDayPicker][ngModel]',
 })
-export class DatePickerDirective implements OnInit, OnChanges {
-  @Input('dpDayPicker') config: IDatePickerDirectiveConfig = {
-    hideInputContainer: true,
-  };
+export class DatePickerDirective implements OnInit {
+  private _config: IDatePickerDirectiveConfig;
+  private _attachTo: ElementRef | string;
```

Comment:
I think this should be only `ElementRef`, strings should be handled before the assignment